recipes/msix/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class MsixConan(ConanFile):
+    name = "msix"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/msix-packaging"
+    description = "An SDK for creating MSIX packages"
+    topics = ("msix", "sdk", "packaging", "conan-recipe")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "crypto_lib": ["crypt32", "openssl"],
+        "pack": [True, False],
+        "skip_bundles": [True, False],
+        "use_external_zlib": [True, False],
```

Comment:
We want to always use external libs.
Using a an included/included vendored zlib will open the possibility of having some symbols available twice.
The linker can then choose symbols from msix or from zlib.

So please remove this option and always use external zlib (=our cci package)

Reply:
This option allows to choose system libraries for compression when building on some platforms:
```
set(COMPRESSION_LIB "zlib")
if(((IOS) OR (MACOS)) AND (NOT USE_EXTERNAL_ZLIB))
    set(COMPRESSION_LIB "libCompression")
elseif((AOSP) AND (NOT USE_EXTERNAL_ZLIB))
    set(COMPRESSION_LIB "inbox zlib")
endif()
```
If this option is `on`, the `zlib` package from CCI will be used.

Reply:
Looks like current master changed the name of the `USE_EXTERNAL_ZLIB` variable already:
https://github.com/microsoft/msix-packaging/blob/a8c86c68f18c7fb641fe5e60f31e86e176f9bd29/cmake/msix_options.cmake#L88-L94

Current master has:
https://github.com/microsoft/msix-packaging/blob/c8af99506ffd0c1513fad39cdadfac281723c3e3/cmake/msix_options.cmake#L16

Reply:
The upstream didn't rename the variable, we did it so that it reflects the idea that when it's `on`, the build will use the external (CCI) package for `zlib`. The original name and description is:
```
option(USE_MSIX_SDK_ZLIB "Use zlib implementation under lib/zlib. If off, uses inbox compression library. For Windows and Linux this is no-opt." OFF)
```
The modified version:
```
option(USE_EXTERNAL_ZLIB "Use an external zlib package. If off, uses inbox compression library. For Windows and Linux this is no-opt." OFF)
```

Reply:
Vendored libraries are often provided for users without package manager as a way to quickly get building without building all dependencies.
We're using conan here with the cci library. So I think it's best to remove this option and unconditionally disable using a vendored zlib.

Reply:
By the term "vendored" we meant libraries whose sources had been made a part of the library, when I wrote about system `zlib` I meant these ones:

https://developer.android.com/ndk/guides/stable_apis#zlib_compression

https://developer.apple.com/documentation/compression

https://github.com/microsoft/msix-packaging/blob/3fbd28851b0ac525324781ec20634a2f0515c0ef/src/msix/CMakeLists.txt#L292_L298

We can omit these possibilities, shall we?

Reply:
What we want to avoid (at all cost) is to use the vendored zlib here: https://github.com/microsoft/msix-packaging/blob/5f977a79d4f2bd189c1d99ae8f501316a282191a/lib/CMakeLists.txt#L22

Does the current combination of options + patches completely avoid that?

Reply:
> Does the current combination of options + patches completely avoid that?

The current patch completely removes processing of the `lib` subdirectory: https://github.com/conan-io/conan-center-index/blob/a3144aa600399cfd7cd535b10d847ed80b658bed/recipes/msix/all/patches/1.7/cmake.patch#L79

conan-center-index/cmake.patch at a3144aa600399cfd7cd535b10d847ed80b658bed · conan-io/conan-center-index
https://github.com

Reply:
As point out the patch delete the usage of this option so it can be removed from the recipe!

```suggestion
```

Reply:
The option hasn't been deleted, [the commit](https://github.com/conan-io/conan-center-index/pull/6475/commits/e050a90f69494cd24755a65778701df358a08588) just reverts it's name to the original one.